### ISSUE TITLE
Segfault for default class variable in crystal 0.11.0

### ISSUE THREAD

**Issue by ysbaddaden** [manastech/crystal#2059]
With Crystal 0.11.0, running the following program with no arguments will segfault when trying to access `Test.logger`. There are two ways to workaround the bug: 1. pass an argument to the program, which will set `Test.logger`, or 2. remove the `Test.logger = Logger.new(file)` line.

```crystal
require "logger"

module Test
  def self.logger
    @@logger ||= Logger.new(STDOUT)
  end

  def self.logger=(@@logger)
  end
end

if ARGV[0]?
  Test.logger = Logger.new(File.open("test_logger.log", "w"))
end

Test.logger.info("ok")
```

Note that in doesn't fail with Crystal 0.10.2.

**Comment by asterite** [comment]
Ouch, that's an ugly regression. My fault, I changed [this line](https://github.com/manastech/crystal/commit/d8ef2ced522c39ba7803f6242aa542bb92b3fed1#diff-01f54d505eee061673ab3cdeeb33f125L544). There was even a spec for this but I decided to remove it because I couldn't find the logic in it. I re-added the spec tied to this issue, and added some comments.

With these regressions a 0.11.1 (or 0.12.0) will come out soon.

**Comment by ysbaddaden** [comment]
Let's say 0.11.0 was a RC, and 0.11.1 is the real thing ;-)

**Comment by asterite** [comment]
Hehe, yes. We actually talked about doing a RC with @waj, maybe we should start doing that.

Another thing that we talked about is having release branches, so we can cherry pick some fixes into 0.11.1 without including other things.

I guess this is normal, 0.11.0 changed and introduced a lot of things.
